packages/sdk/src/browser-wallets/xverse/index.ts

```diff
@@ -1,278 +1,90 @@
-      },
-      message: "Sign PSBT",
-      psbtBase64: psbt.toBase64(),
-      broadcast: false,
-      inputsToSign,
-    },
-    onFinish: handleOnFinish,
-    onCancel: handleOnCancel,
-  };
-
-  await signTransaction(xverseOptions);
-
-  // The Return is supplied by the await statement above, which extracts the hex and optional base64 from the response.
-  // Hex is always returned, hence the not null assertion.
-  // In cases where there is no hex, an error would be thrown by the handleOnFinish function.
-  return { hex: hex!, base64 };
+  return satsConnectWalletSignPsbt(getXverseWalletProvider, psbt, {
+    finalize,
+    extractTx,
+    network,
+    inputsToSign,
+  });
 }
 
-/**
- * Signs a message.
- *
- * @param message Message to be signed
- * @param address Address to sign with
- * @param network Network (mainnet or testnet)
- * @returns An object containing `base64` and `hex`.
- * @throws {BrowserWalletNotInstalledError} Wallet is not installed
- * @throws {BrowserWalletSigningError} Failed to sign with Xverse
- * @throws {OrditSDKError} Invalid options provided
- * @throws {BrowserWalletRequestCancelledByUserError} Request was cancelled by user
- */
 async function signMessage(
   message: string,
   address: string,
   network: BrowserWalletNetwork = "mainnet",
 ): Promise<BrowserWalletSignResponse> {
   if (!isInstalled()) {
-    throw new BrowserWalletNotInstalledError("Xverse not installed");
-  }
-  if (!message || !network || !address) {
-    throw new OrditSDKError("Invalid options provided");
+    throw new BrowserWalletNotInstalledError("Selected wallet not installed");
   }
 
-  let hex: string;
-  let base64: string | null = null;
-
-  const handleOnFinish = (response: XverseSignMessageResponse) => {
-    if (!response) {
-      throw new BrowserWalletSigningError(
-        "Failed to sign message using Xverse",
-      );
-    }
-
-    hex = Buffer.from(response, "base64").toString("hex");
-    base64 = response;
-  };
-
-  const handleOnCancel = () => {
-    throw new BrowserWalletRequestCancelledByUserError();
-  };
-
-  const xverseOptions: XverseSignMessageOptions = {
-    payload: {
-      network: {
-        type: NETWORK_TO_BITCOIN_NETWORK_TYPE[network],
-      },
-      message,
-      address,
-    },
-    onFinish: handleOnFinish,
-    onCancel: handleOnCancel,
-  };
-
-  await xverseSignMessage(xverseOptions);
-
-  // The Return is supplied by the await statement above, which extracts the hex and optional base64 from the response.
-  // Hex is always returned, hence the not null assertion.
-  // In cases where there is no hex, an error would be thrown by the handleOnFinish function.
-  return { hex: hex!, base64 };
+  return satsConnectWalletSignMessage(
+    getXverseWalletProvider,
+    message,
+    address,
+    network,
+  );
 }
 
-export { getAddresses, isInstalled, signMessage, signPsbt };
-
-export * from "../types";
-export * from "./types";
+export {
+  getAddresses,
+  getXverseWalletProvider,
```

Comment:
Will this provider be required for ord-connect? We can remove this export if not required since it doesn't follow the docs

```suggestion
```